versioned/persist/mongodb/src/main/java/org/projectnessie/versioned/persist/mongodb/MongoDatabaseAdapter.java

```diff
@@ -482,4 +518,84 @@ protected RefLog fetchFromRefLog(NonTransactionalOperationContext ctx, Hash refL
       NonTransactionalOperationContext ctx, List<Hash> hashes) {
     return fetchPage(client.getRefLog(), hashes, ProtoSerialization::protoToRefLog);
   }
+
+  @Override
+  protected void writeBinaries(Stream<Entry<BlobKey, BlobValue>> binaries) {
+    binaries
+        .map(
+            e -> {
+              BlobValue value = e.getValue();
+              Document doc = toDoc(blobKeyAsString(e.getKey()), value.toByteArray());
+              if (value.hasHash()) {
+                doc.put(VALUE_HASH_PROPERTY_NAME, value.getHash());
+              }
+              return doc;
+            })
+        .forEach(
+            doc ->
+                client
+                    .getBlobs()
+                    .updateOne(
```

Comment:
Should we check the result of `updateOne` (e.g. `result.wasAcknowledged()`)? I'm not sure Mongo Client will throw exceptions in all cases the update did not happen :thinking:

Reply:
done.